MeshGeoToolsLib/CreateSearchLength.h

```diff
@@ -11,22 +11,25 @@
 
 #pragma once
 
+#include <memory>
+
 #include "BaseLib/ConfigTree.h"
 
 #include "MeshGeoToolsLib/HeuristicSearchLength.h"
 #include "MeshGeoToolsLib/SearchLength.h"
 
 namespace MeshGeoToolsLib
 {
-MeshGeoToolsLib::SearchLength createSearchLengthAlgorithm(
+std::unique_ptr<MeshGeoToolsLib::SearchLength> createSearchLengthAlgorithm(
```

Comment:
Could be, but the actual reason is that HeuristicSearchLength (which is derived from SearchLength) could be returned here and this would be an implicit cast. Pointer or reference type is needed to keep the polymorphic behaviour.